### PR TITLE
Change the release process

### PR DESCRIPTION
**Pull request by lindseydew**
## What does this change?

This changes our release to maven process for running on a developers machine to running on github workflow as per [these instuctions](https://github.com/guardian/gha-scala-library-release-workflow/blob/main/docs/configuration.md)

## Accessibility

-   [ ] [Tested with screen reader](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#screen-reader)
-   [ ] [Navigable with keyboard](https://github.com/guardian/accessibility/blob/main/people-and-technology/02-physical.md#keyboard)
-   [ ] [Colour contrast passed](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#contrast)
-   [ ] [The change doesn't use only colour to convey meaning](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#use-of-colour)

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: build
```
Comment (rtyley) [inline .github/workflows/build.yml:1]: CI?!

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        with:
+          distribution: corretto
+          cache: sbt
+          java-version: 11
+      - name: Build and Test
+        run: sbt -v +test
+      - name: Test Summary
+        uses: test-summary/action@v2
+        with:
+          paths: "test-results/**/TEST-*.xml"
```
Comment (rtyley) [inline .github/workflows/ci.yml:22]: For this to work, we need the build.sbt updated to generate the files, and put them in the right place:

https://github.com/guardian/etag-caching/blob/7ecc04981f5a42a0f2ecb10631f28da571a49836/build.sbt#L18-L19

**Review by rtyley — APPROVED** [review]
Looks great, some small tweaks to check...!

Maybe add something to the readme for the release instructions? eg https://github.com/guardian/flexible-octopus-model/blob/main/README.md

File: .github/workflows/build.yml (hunk 1 above)
Comment (lindseydew) [reply .github/workflows/build.yml]: Renamed

File: .github/workflows/ci.yml (hunk 2 above)
Comment (lindseydew) [reply .github/workflows/ci.yml:22]: Ah thanks, updated the build

File: build.sbt [hunk 3]
```diff
@@ -1,47 +1,21 @@
...
-    commitReleaseVersion,
-    tagRelease,
-    publishArtifacts,
-    releaseStepCommand("sonatypeBundleRelease"),
-    setNextVersion,
-    commitNextVersion,
-    pushChanges
-  )
+ThisBuild / scalacOptions := Seq("-deprecation", "-release:11")
+ThisBuild / scalaVersion := scala_2_12
```
[the remark this answers is not in the document]
Comment (lindseydew) [reply build.sbt]: Yeah good idea

File: build.sbt [hunk 4]
```diff
@@ -1,47 +1,21 @@
...
+  scalaVersion.value,
+  scala_2_13,
 )
+
 val awsSdk2Version = "2.25.13"
 
 lazy val root = (project in file("."))
-  .settings(publishSettings)
   .settings(
     licenses += ("Apache-2.0", url("http://www.apache.org/licenses/LICENSE-2.0.html")),
```
[the remark this answers is not in the document]
Comment (lindseydew) [reply build.sbt]: Thanks, updated

File: build.sbt [hunk 5]
```diff
@@ -56,5 +30,19 @@ lazy val root = (project in file("."))
       "io.netty" % "netty-codec-http2" % "4.1.107.Final",
       "io.netty" % "netty-common" % "4.1.107.Final",
       "org.specs2" %% "specs2-core" % "4.20.5" % "test"
+    ),
+    publish / skip := true,
```
[the remark this answers is not in the document]
Comment (lindseydew) [reply build.sbt]: Thanks, removed

**Pull request merged**